tests/e2e/vmlistener_test.go

```diff
@@ -59,8 +59,10 @@ func (s *VMListenerTestParams) SetUpTest(c *C) {
 }
 
 func (s *VMListenerTestParams) TearDownTest(c *C) {
-	// Note: no need to remove container, it is already removed as
-	// vm got killed and `--restart=always` flag is not added
+	if dockercli.IsContainerExist(s.dockerHostIP, s.containerName) {
```

Comment:
Instead of removing the container here can we run it with the --rm flag so it will be removed by docker when the container exits.